### ISSUE TITLE
Cannot correctly escape in IngressRoute 

### ISSUE THREAD

**Issue by fefe982** [containous/traefik#6773]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I'll trying to add an IngressRoute using `HeadersRegexp` to match a literal `.` (dot) in an IP address. Withe the following config:
```
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: bug
  namespace: traefik
spec:
  entryPoints:
    - web
  routes:
  - match: Host(`dummy.host`) && HeadersRegexp(`X-Real-Ip`, `172\..*`)
    kind: Rule
    services:
    - name: dummy-svc
      namespace: dummy-test
      port: 80
```

### What did you expect to see?

It should be a correct rule and can match `X-Real-Ip: 172.18.3.1`.

### What did you see instead?

1. This rule would not show up in the UI. If there is an error, it should show up in the dashboard. Disappearing from the dashboard made me confused and hard to track what is going on.
2. It doesn't take effect
3. The logs of traefik has the following line:
```
time="2020-05-06T15:12:41Z" level=error msg="Invalid syntax for match rule: Host(`dummy.host`) && HeadersRegexp(`X-Real-Ip`, `172\\..*`)" ingress=bug namespace=traefik providerName=kubernetescrd
```

By the way, I also tried with double backslash ``HeadersRegexp(`X-Real-Ip`, `172\\..*`)``. It is recognized as a correct rule, but it cannot match an IP of `172.18.3.0`.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.2.0
Codename:     chevrotin
Go version:   go1.14.1
Built:        2020-03-25T17:32:57Z
OS/Arch:      linux/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Installed using helm chart `traefik/traefix` with helm3 from "https://containous.github.io/traefik-helm-chart" into a kubernetes cluster.

```toml
# (paste your configuration here)
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```

**Comment by fefe982** [comment]
After some digging into the code, I found that the rule is rejected because it does not pass the check of [`checkStringQuoteValidity`](https://github.com/containous/traefik/blob/5560ab28f2b1525759abb7b117189a53310ed1e9/pkg/provider/kubernetes/crd/kubernetes.go#L612) . The string is put into a pair of double quotes `"` and passed into `strconv.Unquote` to see if there is any error. And probably because `\.` is illegal in golang, it is rejected as invalid.

However, the string can be correctly parsed by `parse.ParseExpr`, which is used internally by [`predicate.(*predicateParser).Parse()`](https://github.com/vulcand/predicate/blob/8fbfb3ab0e94276b6b58bec378600829adc7a203/parse.go#L24). `predicate.(*predicateParser).Parse()` is used in [`AddRoute`](https://github.com/containous/traefik/blob/5560ab28f2b1525759abb7b117189a53310ed1e9/pkg/rules/rules.go#L46) to parse the rule. It parses the string a golang source piece, not in double quotes. Here the `\` appears in backtick and has no special meaning.

I tried it in with `parse.ParseExpr` in go playground and it return no error.

(I did not compile or run the code, just read it on the github)

**Comment by fefe982** [comment]
I'm using ``HeadersRegexp(`X-Real-Ip`, `172[.].*`)`` as a workaround now.

**Comment by fefe982** [comment]
Get a clone of the code, and commented out the offending check. After that the configuration (``eadersRegexp(`X-Real-Ip`, `172\..*`)`` can work as expected.

**Comment by dtomcej** [comment]
Hi @fefe982,

Thanks for this report! The original source of that code came from here (which I wrote):

https://github.com/containous/traefik/pull/3964

In Traefik v1.x, an internal template was used to transfer configurations from various providers to the server core.

The issue encountered was that the kubernetes API returned a string literal, which we then used as a double quoted string. The issue encountered was that values returned from kubernetes could be non-valid double quoted strings, which broke the internal template, causing the server core to stop processing.

Checking the quotability of these values prevented an erroneous value from breaking the server core.

Traefik v2 does not use an internal template. As such, this check should not be required, and should be removable.

Also as an aside, It should be possible to pass double quoted (escaped) strings into the host rules, instead of string literals:

```
Host("dummy.host") && HeadersRegexp("X-Real-Ip", "172\\..*")
```

However, I'm not sure offhand if the double quoted regex would pass the unneeded unquote test.
